sqlalchemy_utils/functions/database.py

```diff
@@ -542,7 +538,7 @@ def create_database(url, encoding='utf8', template=None):
     dialect_name = url.get_dialect().name
     dialect_driver = url.get_dialect().driver
 
-    if dialect_name == 'postgres':
+    if dialect_name == 'postgres' or dialect_name == 'postgresql':
```

Comment:
Do you have evidence this is needed? This is not coming from the connection string, but after SQLAlchemy processing, which always uses `postgres`.

Reply:
Yeah, I figured something was weird about this.

But yes, I did need it. See this from my local testing(s):
```sh
adstakr/tests/conftest.py:64:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy_utils/functions/database.py:556: in create_database
    url = _set_url_database(url, database=None)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

url = postgresql://adstakr:***@localhost/adstakr_test, database = None

    def _set_url_database(url: sa.engine.url.URL, database):
        # try:
        #     ret = sa.engine.URL.create(
        #         drivername=url.drivername,
        #         username=url.username,
        #         password=url.password,
        #         host=url.host,
        #         port=url.port,
        #         database=database,
        #         query=url.query
        #     )
        # except AttributeError:  # SQLAlchemy <1.4
        #     url.database = database
        #     ret = url
        # breakpoint()
        if hasattr(url, 'set'):
            ret = url.set(database=copy(database))
        else:
            url.database = copy(database)
            ret = url
>       assert ret.database == database, f"Failed to set url database: {ret.database} != {database}"
E       AssertionError: Failed to set url database: adstakr_test != None

../.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy_utils/functions/database.py:444: AssertionError
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /home/bradenmars/projects/work/adstakr/adstakr-api/.direnv/python-3.8.8/lib/python3.8/site-packages/sqlalchemy_utils/functions/database.py(444)_set_url_database()
-> assert ret.database == database, f"Failed to set url database: {ret.database} != {database}"
(Pdb) l
439         if hasattr(url, 'set'):
440             ret = url.set(database=copy(database))
441         else:
442             url.database = copy(database)
443             ret = url
444  ->     assert ret.database == database, f"Failed to set url database: {ret.database} != {database}"
445         return ret
446
447
448     def _get_scalar_result(engine, sql):
449         with engine.connect() as conn:
(Pdb) url
postgresql://adstakr:***@localhost/adstakr_test
(Pdb) url.get_dialect().name
'postgresql'
(Pdb) sa.__version__
'1.4.3'
(Pdb)
```

Reply:
You cannot use `postgresql` in connection strings in SQLAlchemy 1.4 any more, see e.g. https://github.com/sqlalchemy/sqlalchemy/issues/6083

Reply:
Huh. Okay haha, bare with me here... I am about on the edge of falling asleep right now.

But aren't they saying its the other way around?
See: https://github.com/sqlalchemy/sqlalchemy/blob/8fc5005dfe3eb66a46470ad8a8c7b95fc4d6bdca/lib/sqlalchemy/dialects/postgres.py

As in,
`postgres` :x: 
`postgresql: :heavy_check_mark: 

Correct me if I'm being stupid haha.

Reply:
No, you're right, I'm sorry, it's one way in the connection string and the other internally!